### PR TITLE
MAINT: Big Renaming

### PR DESCRIPTION
**Pull request by hhslepicka**
## Description

Renaming of the package from `typhon` to `typhos` while keeping backwards compatibility.

## Motivation and Context

Due to a naming conflict at PyPI and the request from the Pilot Project to have the packages at PyPI we decided to rename Typhon to Typhos.

## How Has This Been Tested?

The tests were unchanged and they are passing

## Where Has This Been Documented?

Docs were updated.

## IMPORTANT:
Unfortunately the only piece that we could not offer backwards compatibility was at Qt Stylesheets...
If you use them, please update the entry or add also the Typhos entry to keep it working properly.

## TODO:

- [x] Rename typhon -> typhos at tests
- [x] Add at least two tests for smoketest of backwards compatibility
- [x] Add entry at docs main page and README.md to notify of the renaming

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -175,8 +175,8 @@
 # (source start file, target name, title, author,
 #  dir menu entry, description, category)
 texinfo_documents = [
-    (master_doc, 'Typhon', 'Typhon Documentation',
-     author, 'Typhon', 'One line description of project.',
+    (master_doc, 'Typhos', 'Typhos Documentation',
+     author, 'Typhos', 'One line description of project.',
```
Comment (klauer) [inline docs/source/conf.py]:
```suggestion
     author, 'Typhos', 'Typhos Documentation',
```

**Comment by hhslepicka** [comment]
Just for the future...
The tests are passing when using `typhon` for imports...
<details>
<blockquote>
$ coverage run run_tests.py --show-capture=no
pytest arguments: ['-v', '-vrxs', '--show-capture=no']
============================================================================================================================= test session starts ==============================================================================================================================
platform darwin -- Python 3.7.3, pytest-5.3.1, py-1.8.0, pluggy-0.13.0 -- /Users/slepicka/mc/envs/py37/bin/python
cachedir: .pytest_cache
PyQt5 5.12.3 -- Qt runtime 5.12.5 -- Qt compiled 5.12.5
rootdir: /Users/slepicka/sandbox/git-pcdshub/typhon-git, inifile: pytest.ini
plugins: timeout-1.3.3, qt-3.3.0, cov-2.8.1
collected 92 items

tests/test_cli.py::test_cli_version PASSED                                                                                                                                                                                                                               [  1%]
tests/test_cli.py::test_cli_happi_cfg PASSED                                                                                                                                                                                                                             [  2%]
tests/test_cli.py::test_cli_bad_entry PASSED                                                                                                                                                                                                                             [  3%]
tests/test_cli.py::test_cli_no_entry PASSED                                                                                                                                                                                                                              [  4%]
tests/test_cli.py::test_cli_stylesheet PASSED                                                                                                                                                                                                                            [  5%]
tests/test_console.py::test_base_console PASSED                                                                                                                                                                                                                          [  6%]
tests/test_console.py::test_add_device XPASS                                                                                                                                                                                                                             [  7%]
tests/test_designer.py::test_typhon_panel_plugin_smoke PASSED                                                                                                                                                                                                            [  8%]
tests/test_display.py::test_device_display PASSED                                                                                                                                                                                                                        [  9%]
tests/test_display.py::test_display_without_md PASSED                                                                                                                                                                                                                    [ 10%]
tests/test_display.py::test_display_with_md PASSED                                                                                                                                                                                                                       [ 11%]
tests/test_display.py::test_display_type_change PASSED                                                                                                                                                                                                                   [ 13%]
tests/test_display.py::test_display_modified_templates PASSED                                                                                                                                                                                                            [ 14%]
tests/test_display.py::test_display_force_template PASSED                                                                                                                                                                                                                [ 15%]
tests/test_display.py::test_display_with_channel PASSED                                                                                                                                                                                                                  [ 16%]
tests/test_display.py::test_display_device_class_property PASSED                                                                                                                                                                                                         [ 17%]
tests/test_display.py::test_display_device_name_property PASSED                                                                                                                                                                                                          [ 18%]
tests/test_display.py::test_display_with_py_file PASSED                                                                                                                                                                                                                  [ 19%]
tests/test_func.py::test_func_display_creation PASSED                                                                                                                                                                                                                    [ 20%]
tests/test_func.py::test_func_execution PASSED                                                                                                                                                                                                                           [ 21%]
tests/test_func.py::test_func_exceptions PASSED                                                                                                                                                                                                                          [ 22%]
tests/test_func.py::test_func_panel PASSED                                                                                                                                                                                                                               [ 23%]
tests/test_func.py::test_method_button_execute PASSED                                                                                                                                                                                                                    [ 25%]
tests/test_func.py::test_method_button_use_status PASSED                                                                                                                                                                                                                 [ 26%]
tests/test_func.py::test_func_docstrings PASSED                                                                                                                                                                                                                          [ 27%]
tests/test_log.py::test_log_display PASSED                                                                                                                                                                                                                               [ 28%]
tests/test_plot.py::test_add_signal PASSED                                                                                                                                                                                                                               [ 29%]
tests/test_plot.py::test_curve_methods PASSED                                                                                                                                                                                                                            [ 30%]
tests/test_plot.py::test_curve_creation_button PASSED                                                                                                                                                                                                                    [ 31%]
tests/test_plot.py::test_device_plot PASSED                                                                                                                                                                                                                              [ 32%]
tests/test_positioner.py::test_positioner_widget_no_limits PASSED                                                                                                                                                                                                        [ 33%]
tests/test_positioner.py::test_positioner_widget_fixed_limits PASSED                                                                                                                                                                                                     [ 34%]
tests/test_positioner.py::test_positioner_widget_with_signal_limits PASSED                                                                                                                                                                                               [ 35%]
tests/test_positioner.py::test_positioner_widget_readback PASSED                                                                                                                                                                                                         [ 36%]
tests/test_positioner.py::test_positioner_widget_stop PASSED                                                                                                                                                                                                             [ 38%]
tests/test_positioner.py::test_positioner_widget_set PASSED                                                                                                                                                                                                              [ 39%]
tests/test_positioner.py::test_positioner_widget_positive_tweak PASSED                                                                                                                                                                                                   [ 40%]
tests/test_positioner.py::test_positioner_widget_negative_tweak PASSED                                                                                                                                                                                                   [ 41%]
tests/test_positioner.py::test_positioner_widget_moving_property PASSED                                                                                                                                                                                                  [ 42%]
tests/test_positioner.py::test_positioner_widget_last_move PASSED                                                                                                                                                                                                        [ 43%]
tests/test_signal.py::test_panel_creation PASSED                                                                                                                                                                                                                         [ 44%]
tests/test_signal.py::test_panel_add_enum PASSED                                                                                                                                                                                                                         [ 45%]
tests/test_signal.py::test_add_dead_signal PASSED                                                                                                                                                                                                                        [ 46%]
tests/test_signal.py::test_add_pv PASSED                                                                                                                                                                                                                                 [ 47%]
tests/test_signal.py::test_typhon_panel PASSED                                                                                                                                                                                                                           [ 48%]
tests/test_signal.py::test_typhon_panel_sorting PASSED                                                                                                                                                                                                                   [ 50%]
tests/test_signal.py::test_signal_widget_waveform PASSED                                                                                                                                                                                                                 [ 51%]
tests/test_signal.py::test_signal_widget_image PASSED                                                                                                                                                                                                                    [ 52%]
tests/test_status.py::test_previously_done_status_in_thread PASSED                                                                                                                                                                                                       [ 53%]
tests/test_status.py::test_status_finished_during_lag PASSED                                                                                                                                                                                                             [ 54%]
tests/test_status.py::test_status_thread_completed PASSED                                                                                                                                                                                                                [ 55%]
tests/test_status.py::test_status_thread_timeout PASSED                                                                                                                                                                                                                  [ 56%]
tests/test_suite.py::test_suite_with_child_devices PASSED                                                                                                                                                                                                                [ 57%]
tests/test_suite.py::test_suite_without_children PASSED                                                                                                                                                                                                                  [ 58%]
tests/test_suite.py::test_suite_tools PASSED                                                                                                                                                                                                                             [ 59%]
tests/test_suite.py::test_suite_get_subdisplay_by_device PASSED                                                                                                                                                                                                          [ 60%]
tests/test_suite.py::test_suite_subdisplay_parentage PASSED                                                                                                                                                                                                              [ 61%]
tests/test_suite.py::test_suite_get_subdisplay_by_name PASSED                                                                                                                                                                                                            [ 63%]
tests/test_suite.py::test_suite_show_display_by_device PASSED                                                                                                                                                                                                            [ 64%]
tests/test_suite.py::test_suite_show_display_by_parameter PASSED                                                                                                                                                                                                         [ 65%]
tests/test_suite.py::test_suite_hide_subdisplay_by_device PASSED                                                                                                                                                                                                         [ 66%]
tests/test_suite.py::test_suite_hide_subdisplay_by_parameter PASSED                                                                                                                                                                                                      [ 67%]
tests/test_suite.py::test_suite_hide_subdisplays PASSED                                                                                                                                                                                                                  [ 68%]
tests/test_suite.py::test_device_parameter_tree PASSED                                                                                                                                                                                                                   [ 69%]
tests/test_suite.py::test_suite_embed_device PASSED                                                                                                                                                                                                                      [ 70%]
tests/test_suite.py::test_suite_embed_device_by_name PASSED                                                                                                                                                                                                              [ 71%]
tests/test_suite.py::test_hide_embedded_display PASSED                                                                                                                                                                                                                   [ 72%]
tests/test_suite.py::test_suite_save_util PASSED                                                                                                                                                                                                                         [ 73%]
tests/test_suite.py::test_suite_save PASSED                                                                                                                                                                                                                              [ 75%]
tests/test_suite.py::test_suite_save_cancel_smoke PASSED                                                                                                                                                                                                                 [ 76%]
tests/test_utils.py::test_clean_name PASSED                                                                                                                                                                                                                              [ 77%]
tests/test_utils.py::test_stylesheet PASSED                                                                                                                                                                                                                              [ 78%]
tests/test_utils.py::test_grab_kind PASSED                                                                                                                                                                                                                               [ 79%]
tests/test_utils.py::test_qtdesigner_env SKIPPED                                                                                                                                                                                                                         [ 80%]
tests/test_utils.py::test_typhonbase_repaint_smoke PASSED                                                                                                                                                                                                                [ 81%]
tests/test_utils.py::test_raise_to_operator_msg PASSED                                                                                                                                                                                                                   [ 82%]
tests/test_utils.py::test_load_suite PASSED                                                                                                                                                                                                                              [ 83%]
tests/test_utils.py::test_load_suite_with_bad_py_file PASSED                                                                                                                                                                                                             [ 84%]
tests/test_widgets.py::test_sidebar_item PASSED                                                                                                                                                                                                                          [ 85%]
tests/test_widgets.py::test_signal_dialog_button_show PASSED                                                                                                                                                                                                             [ 86%]
tests/test_widgets.py::test_signal_dialog_button_repeated_show PASSED                                                                                                                                                                                                    [ 88%]
tests/test_widgets.py::test_dialog_button_instances_smoke[Waveform] PASSED                                                                                                                                                                                               [ 89%]
tests/test_widgets.py::test_dialog_button_instances_smoke[Image] PASSED                                                                                                                                                                                                  [ 90%]
tests/plugins/test_core.py::test_signal_connection PASSED                                                                                                                                                                                                                [ 91%]
tests/plugins/test_core.py::test_metadata PASSED                                                                                                                                                                                                                         [ 92%]
tests/plugins/test_core.py::test_disconnection PASSED                                                                                                                                                                                                                    [ 93%]
tests/plugins/test_core.py::test_array_signal_send_value PASSED                                                                                                                                                                                                          [ 94%]
tests/plugins/test_core.py::test_array_signal_put_value PASSED                                                                                                                                                                                                           [ 95%]
tests/plugins/test_happi.py::test_connection PASSED                                                                                                                                                                                                                      [ 96%]
tests/plugins/test_happi.py::test_connection_for_child PASSED                                                                                                                                                                                                            [ 97%]
tests/plugins/test_happi.py::test_bad_address_smoke PASSED                                                                                                                                                                                                               [ 98%]
tests/plugins/test_happi.py::test_happi_is_optional PASSED                                                                                                                                                                                                               [100%]

=============================================================================================================================== warnings summary ===============================================================================================================================
tests/test_console.py::test_base_console
  /Users/slepicka/mc/envs/py37/lib/python3.7/site-packages/jupyter_client/session.py:371: DeprecationWarning: Session._key_changed is deprecated in traitlets 4.1: use @observe and @unobserve instead.
    def _key_changed(self):

tests/test_console.py::test_base_console
  /Users/slepicka/mc/envs/py37/lib/python3.7/site-packages/jupyter_client/threaded.py:183: DeprecationWarning: zmq.eventloop.ioloop is deprecated in pyzmq 17. pyzmq now works with default tornado and asyncio eventloops.
    self.ioloop = ioloop.IOLoop()

tests/test_display.py::test_device_display
tests/test_widgets.py::test_dialog_button_instances_smoke[Image]
  /Users/slepicka/mc/envs/py37/lib/python3.7/asyncio/base_events.py:623: ResourceWarning: unclosed event loop <_UnixSelectorEventLoop running=False closed=False debug=False>
    source=self)

tests/test_signal.py::test_typhon_panel
tests/test_signal.py::test_typhon_panel_sorting
  /Users/slepicka/sandbox/git-slaclab/pydm-git/pydm/application.py:456: UserWarning: 'PyDMApplication.establish_widget_connections' is deprecated, this function is now found on `utilities.establish_widget_connections`.
    warnings.warn("'PyDMApplication.establish_widget_connections' is deprecated, "

tests/test_suite.py::test_suite_save
  /Users/slepicka/sandbox/git-pcdshub/typhon-git/tests/test_suite.py:164: ResourceWarning: unclosed file <_io.TextIOWrapper name='/var/folders/f9/f1fnxmpd3nb1w5xvcf84_swxs5r5tq/T/test.py' mode='r' encoding='UTF-8'>
    assert str(devices) in open(str(tfile), 'r').read()

tests/test_utils.py::test_stylesheet
  /Users/slepicka/mc/envs/py37/lib/python3.7/site-packages/qdarkstyle/__init__.py:388: PendingDeprecationWarning: load_stylesheet_pyqt5() will be deprecated in version 3,set QtPy environment variable to specify the Qt binding and use load_stylesheet()
    PendingDeprecationWarning

tests/plugins/test_happi.py::test_happi_is_optional
  /Users/slepicka/mc/envs/py37/lib/python3.7/importlib/__init__.py:169: DeprecationWarning: WARNING: typhon was renamed to typhos along with all classes and methods that started with typhon or Typhon.
    _bootstrap._exec(spec, module)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================================================================================================================== short test summary info ============================================================================================================================
SKIPPED [1] tests/test_utils.py:65: Package not installed via CONDA
============================================================================================================ 90 passed, 1 skipped, 1 xpassed, 9 warnings in 21.51s =============================================================================================================
</blockquote>
</details>

**Comment by ZLLentz** [comment]
Should we rename this repo right before we merge this PR?

**Comment by hhslepicka** [comment]
@ZLLentz I think that we should...
I think that GitHub will still perform the proper redirects if the previous name is used... but I am not sure.

**Comment by klauer** [comment]
I think it will redirect, but we'll see...

But also if we don't have time to get this merged soon, we should put a dummy package on pypi to take the name `typhos`...

**Comment by hhslepicka** [comment]
I am finishing the last two items at the TODO list and I will work on getting Travis happy again.
Should be ready today.

**Comment by hhslepicka** [comment]
All green for now...
A couple of notes here:
- Disabled tests at the conda-recipe due to the fact that it was picking up Python 3.7.6 even if I forced at the `meta.yml` to be less than 3.7.6.
- Added the pip tests at the Travis matrix to the `allow_failures` due to segfault that is happening at the cleanup with ophyd 1.4.0rc3 that I can't reproduce locally and the tests there are passing. Seems to be related to `periodic_update` thread at the `SynPeriodicSignal`.

@klauer and @ZLLentz this is good for review...

**Comment by klauer** [comment]
💯

**Pull request merged**